internal/common/mqtt/mqttconnection.go

```diff
@@ -97,15 +97,11 @@ func StartMQTTClient(brokerURL string, appID string, mqttPort uint) string {
 		return ""
 	}
 	//new client object is created to add to MQTTClient and connection is established and added to URLData
-	clientConfig, err := NewClient(
+	clientConfig, _ := NewClient(
```

Comment:
Why do you remove `err` variable?

Reply:
In case of an error, the log.Panic is triggered, so the variant with an error is not possible in this case.

Reply:
Please exclude the code related to Panic from this PR.
IMHO, it is somewhat severe to call panic() since CloudSync is only one of features of edge orchestration. I'd like to send an error message to the caller.

Reply:
Please exclude the code related to Panic from this PR.
IMHO, it is somewhat severe to call panic() since CloudSync is only one of features of edge orchestration. I'd like to send an error message to the caller.

Reply:
From the security side, any errors in the settings should end with the end of the Edge-Orchestration (Panic). Especially in this case since it is the setting up of keys and certificates. What do you think about it?

Reply:
What about disabling the CloudSync function in this case instead of panic?

Reply:
This does not make sense since the absent of keys and certificates will lead to the inoperability of the internal exchange between nodes and this is independent of the CloudSync function.

Reply:
At the moment, the same keys and certificates are used for the TLS connection.

Reply:
It means there is no panic while [tlsserver](https://github.com/lf-edge/edge-home-orchestration-go/blob/acdae3b29724fa2eade57764350665766fca91f3/internal/restinterface/route/tlsserver/tlsserver.go#L77) is running. Right?
I want to check this because there will be many requests for NewClient.
What about setting tlsconfig only once during [initiation](https://github.com/lf-edge/edge-home-orchestration-go/blob/acdae3b29724fa2eade57764350665766fca91f3/internal/common/mqtt/mqttconfig.go#L95)?

Reply:
It means there is no panic while [tlsserver](https://github.com/lf-edge/edge-home-orchestration-go/blob/acdae3b29724fa2eade57764350665766fca91f3/internal/restinterface/route/tlsserver/tlsserver.go#L77) is running. Right?

I will change it soon! Thank you!

Reply:
> What about setting tlsconfig only once during [initiation](https://github.com/lf-edge/edge-home-orchestration-go/blob/acdae3b29724fa2eade57764350665766fca91f3/internal/common/mqtt/mqttconfig.go#L95)?

I considered this option (I also like it) but:
1) need new architecture for tls (possible to implement in securemgr)
2) perhaps we will have different certificates for internal exchange and external
Therefore, I think to consider it later

Reply:
> It means there is no panic while [tlsserver](https://github.com/lf-edge/edge-home-orchestration-go/blob/acdae3b29724fa2eade57764350665766fca91f3/internal/restinterface/route/tlsserver/tlsserver.go#L77) is running. Right?
> 
> I will change it soon! Thank you!

Done!